lib/spack/docs/contribution_guide.rst

```diff
@@ -203,7 +203,17 @@ to update them.
       ImportError: No module named pkg_resources
 
    that means Flake8 couldn't find setuptools in your ``PYTHONPATH``.
+   
+   If running ``spack flake8`` produces the following message
+   
+   .. code-block:: console
 
+      ==> Error: spack requires 'flake8'. Make sure it is in your path.
+   
+   , and you have installed ``py-flake8``, then try loading the ``py-flake8`` module
+   (``module av py-flake8`` to find the suffix on modules name,
+   then ``module load py-flake8<suffix>``), and re-running ``spack flake8``.
```

Comment:
You should actually be loading `python`, not `py-flake8`. `py-flake8` only puts itself in your `PYTHONPATH`, it doesn't put `py-setuptools` in your `PYTHONPATH`. Activating and loading Python is probably the easiest way, although you can also load `py-flake8`, `py-setuptools`, and `python` without any need for activation.

Basically, Spack searches your `PATH` for `flake8`. Installing a package with Spack doesn't add it to your `PATH`, you have to do that manually (possibly with `module load`). And `py-flake8` has dependencies that need to be in your `PYTHONPATH` for it to run properly.

Reply:
Of course, if you don't have `flake8` installed on the system and you try loading the system Python, you still don't have `flake8`. But if you activate `flake8` and load the corresponding `python` module, everything works out.

Reply:
That makes sense.

Jumping to a rhel7 system below for the examples...
Activating `py-flake8` with a system installed python fails to [create symlinks](https://spack.readthedocs.io/en/latest/basic_usage.html#activating-extensions-globally) without write permissions into the python install dir.  So, the alternative is to load the `py-flake8` package which will set the `PYTHONPATH` which in turn lets the system python find it (as described [here](https://spack.readthedocs.io/en/latest/basic_usage.html?highlight=load%20python#using-extensions).

I'm sure I'm still missing some understanding of what you described, but this seems like a decent start.

```
 $ spack location -i python@2.7.5
/usr
$ spack find -ldv py-flake8
==> 1 installed packages.
-- linux-rhel7-x86_64 / gcc@7.3.0 -------------------------------
romj65i    py-flake8@3.5.0
ifng5lp        ^py-configparser@3.5.0
xevve4z            ^python@2.7.5+dbm~optimizations patches=123082ab3483ded78e86d7c809e98a804b3465b4683c96bd79a2fd799f572244 +pic+pythoncmd+shared~tk~ucs4
jan5tlj        ^py-enum34@1.1.6
aslpzzm        ^py-mccabe@0.6.1
wus5dit        ^py-pycodestyle@2.3.1
fclotm4            ^py-setuptools@39.2.0
lagdt5t        ^py-pyflakes@1.6.0

$ spack activate py-flake8
==> Activating extension py-flake8@3.5.0%gcc@7.3.0 arch=linux-rhel7-x86_64 /romj65i for python@2.7.5%gcc@7.3.0+dbm~optimizations patches=123082ab3483ded78e86d7c809e98a804b3465b4683c96bd79a2fd799f572244 +pic+pythoncmd+shared~tk~ucs4 arch=linux-rhel7-x86_64 /xevve4z
==> Activating extension py-configparser@3.5.0%gcc@7.3.0 arch=linux-rhel7-x86_64 /ifng5lp for python@2.7.5%gcc@7.3.0+dbm~optimizations patches=123082ab3483ded78e86d7c809e98a804b3465b4683c96bd79a2fd799f572244 +pic+pythoncmd+shared~tk~ucs4 arch=linux-rhel7-x86_64 /xevve4z
==> Error: [Errno 13] Permission denied
```

Reply:
Loading `py-flake8` isn't sufficient, you also need to load its dependencies. For example, if you don't load `py-setuptools`, you will see the error message mentioned in this note. In your case, your system Python already had `py-setuptools` installed, which is why you didn't run into the problem.